### PR TITLE
OE-825: Add pagination with infinite scrolling

### PR DESCRIPTION
**Pull request by Toxiapo**
This RP adds the infinite scrolling feature for the collection list.

### CODE REVIEW

File: apps/oeh/src/components/Collection/BookListView.tsx [hunk 1]
```diff
@@ -1,16 +1,86 @@
...
+  );
+
+  if (error) {
+    console.error(error);
+  }
+
+  const isFetchingInitialData = !data && !error;
+  const lastItem = data && data[data.length - 1];
+  const hasMore = !!lastItem?.nextPageUrl;
+  const isFetchingMore = !!(!error && hasMore && size > (data?.length ?? 0));
```
Comment (samanthaandrews) [inline apps/oeh/src/components/Collection/BookListView.tsx:47]: I like how concise this is but it is also hard to read. Can we make it slightly more verbose and increase readability for future us?

File: apps/oeh/src/components/Collection/BookListView.tsx [hunk 2]
```diff
@@ -1,16 +1,86 @@
...
+  collection: CollectionData;
   baseUrl: string;
 };
-const BookListView: React.FC<BookListViewProps> = ({ books, baseUrl }) => {
+
+/**
+ * Currying function used for the useSWRInfinite hook to fetch a collection with
+ * multiple pages.
+ */
+const getFetchKey = (collectionUrl: string, token: string | undefined) => {
```
Comment (samanthaandrews) [inline apps/oeh/src/components/Collection/BookListView.tsx:21]: nice!

**Review by samanthaandrews — CHANGES_REQUESTED** [review]
Just one small thing, otherwise this looks great!

**Comment by samanthaandrews** [comment]
@Toxiapo If you pull in `main` then I think that `OEW E2E Playwright Test` CI check should go away. I changed those tests a few weeks ago to only run on a daily cron.

Not sure why the OEH Playwright test failed...

File: apps/oeh/src/components/Collection/BookListView.tsx (hunk 1 above)
Comment (Toxiapo) [reply apps/oeh/src/components/Collection/BookListView.tsx:51]: I added some comments, let me know if this is better!

**Comment by Toxiapo** [comment]
> @Toxiapo If you pull in `main` then I think that `OEW E2E Playwright Test` CI check should go away. I changed those tests a few weeks ago to only run on a daily cron.
> 
> Not sure why the OEH Playwright test failed...

Yup, it looks like that fixes the issue. Thanks! @samanthaandrews

File: apps/oeh/src/chakra/components/BookCard.tsx [hunk 3]
```diff
@@ -16,7 +32,7 @@ const BookCard = React.forwardRef<
    * encode the book url.
    */
   const encodedBookUrl = encodeURIComponent(encodeURIComponent(book.url));
-  const bookHref = `${baseUrl}/book/${encodedBookUrl}`;
+  const bookHref = `${getRootRoute(baseUrl)}/book/${encodedBookUrl}`;
```
Comment (kristojorg) [inline apps/oeh/src/chakra/components/BookCard.tsx:35]: Might be easier to remove these changes or they will conflict with my changes using `useTabPath`. You could also rebase this on top of your other PR to bring those changes in if you want

File: apps/oeh/src/components/Collection/BookListView.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  // When the page number is greater than the data length, we know the request has been initalized
+  // but not yet fulfilled.
+  const isFetchingMore = !!(!error && hasMore && size > (data?.length ?? 0));
+  const isFetching = isFetchingInitialData || isFetchingMore;
+
+  const books =
+    data?.reduce<OpenEBook[]>(
+      (total, current) => [...total, ...(current.books ?? [])],
+      []
+    ) ?? [];
```
Comment (kristojorg) [inline apps/oeh/src/components/Collection/BookListView.tsx:58]: This formatting seems odd, did prettier get to run on this file?

**Review by kristojorg — APPROVED** [review]
Looks great!

File: apps/oeh/src/chakra/components/BookCard.tsx (hunk 3 above)
Comment (Toxiapo) [reply apps/oeh/src/chakra/components/BookCard.tsx:35]: Good idea!

File: apps/oeh/src/components/Collection/BookListView.tsx (hunk 4 above)
Comment (Toxiapo) [reply apps/oeh/src/components/Collection/BookListView.tsx:58]: It does seem odd, but somehow the prettier likes this format 🤨

**Review by samanthaandrews — APPROVED** [review]
🚀

**Comment by samanthaandrews** [comment]
@Toxiapo Lmk if i can merge this today

**Comment by Toxiapo** [comment]
Good to go!

**Pull request merged**